packages/block-library/src/gallery/block.json

```diff
@@ -3,56 +3,19 @@
 	"name": "core/gallery",
 	"category": "media",
 	"attributes": {
-		"images": {
+		"imageUploads": {
 			"type": "array",
 			"default": [],
-			"source": "query",
-			"selector": ".blocks-gallery-item",
-			"query": {
-				"url": {
-					"type": "string",
-					"source": "attribute",
-					"selector": "img",
-					"attribute": "src"
-				},
-				"fullUrl": {
-					"type": "string",
-					"source": "attribute",
-					"selector": "img",
-					"attribute": "data-full-url"
-				},
-				"link": {
-					"type": "string",
-					"source": "attribute",
-					"selector": "img",
-					"attribute": "data-link"
-				},
-				"alt": {
-					"type": "string",
-					"source": "attribute",
-					"selector": "img",
-					"attribute": "alt",
-					"default": ""
-				},
-				"id": {
-					"type": "string",
-					"source": "attribute",
-					"selector": "img",
-					"attribute": "data-id"
-				},
-				"caption": {
-					"type": "string",
-					"source": "html",
-					"selector": ".blocks-gallery-item__caption"
-				}
+			"items": {
+				"type": "object"
 			}
 		},
-		"ids": {
+		"shortCodeTransforms": {
```

Comment:
What is this about?

Reply:
When transforming a shortcode there is no `transform` method available, so no opportunity to create the innerBlocks as part of the transform process. The only workaround I could see for this was to added the data to a temporary attribute and process this in the editor when detected and then remove the attribute. Let me know if there is a way that I have overlooked to take the image ids from a shortcode and create the innerBlocks as part of the transform process instead.

Reply:
This is the same as above I guess, so there's already a solution in the image block itself.

Reply:
In the case of the gallery short codes all we have is an id, and in the absence of a `transform` method there is no way to convert that id into a url  - i did just try handing an id only to the Image block but it requires a url of some form

Reply:
you're right. It seems it's different than upload. though, I wonder if we should apply a similar technique: 

If we detect  an image block with an id but no url, triggers some effect in the image edit to get the remaining attributes right. We may need a "loading" state in the placeholder or something.

That said, the current approach is not that bad, it's just that attributes are meant to persist and this is changing that a bit..

Reply:
I have had another look at this and the issue is that a shortcode transform provides no opportunity to return a block as there is no `transform` method as part of the transformation as noted here - https://github.com/WordPress/gutenberg/issues/17758. 

If there was a `transform` method with shortcode transformations we could do as you suggested and just create the Image innerBlocks with an id only and let the Image block handle getting the url, etc. but currently the only way I can see to do this would be a two stage transform, eg. go from `shortcode` to `core/shortcode` block and then from `core/shortcode` to `core/gallery` ... but that seems like a messy user experience. Do you know why it is that shortcode transformations have no option to return a block from a `transform` method?

Reply:
@ellatrix would know more here but I guess  the response is that the schema was sufficient for the transforms we had to support so far

Reply:
Sorry for the late reply, going through a bunch of notifications. Maybe there was never a need for it so far? I don't mind if we add a `transform` option.